MidasFrontend/ArduinoHelper.cxx

```diff
@@ -0,0 +1,195 @@
+            gantry_vel_min_mm_s, gantry_vel_max_mm_s);
+        return false;
+    }
+
+    return true;
+}
+
+static int32_t mm_to_cts(float val_mm) {
+    return round(val_mm / mm_cts_ratio);
+}
+
+static uint32_t mm_to_steps(float val_mm) {
+    return round(val_mm / mm_steps_ratio);
+}
+
+static AxisDirection get_direction(int32_t displacement) {
+    return (displacement < 0 ? AXIS_DIR_NEGATIVE : AXIS_DIR_POSITIVE);
+}
+
+static bool handle_serial_result(SerialResult res) {
+    if (res == SERIAL_OK) return true;
+
+    printf("Serial Error: %d\n", res);
+    return false;
+}
+
+static bool handle_axis_result(AxisResult res) {
+    if (res == AXIS_OK) return true;
+
+    printf("Axis Error: %d\n", res);
+    return false;
+}
+
+static bool attempt_move_axis(AxisId axis, int32_t cur_pos_counts, float dest_mm, float vel_mm_s)
+{
+    // Target position
+    int32_t target_counts = mm_to_cts(dest_mm);
+
+    // Relative displacement
+    int32_t disp_counts = (target_counts - cur_pos_counts);
+
+    // Direction
+    AxisDirection dir = get_direction(disp_counts);
+
+    // Velocity
+    uint32_t vel_steps_s = mm_to_steps(vel_mm_s);
+
+    // Send command
+    SerialResult ser_res;
+    AxisResult axis_res;
+    ser_res = comm.move(axis, dir, vel_steps_s, abs(disp_counts), &axis_res, MSG_RECEIVE_TIMEOUT);
+
+    // Handle results
+    return (handle_serial_result(ser_res) && handle_axis_result(axis_res));
+}
+
+/*****************************************************************************/
+/*                             PUBLIC FUNCTIONS                              */
+/*****************************************************************************/
+
+/**
+ * @brief Establishes a serial connection to the Arduino
+ * 
+ * This includes opening the serial device as well as waiting to receive a
+ * ping message from the Arduino to validate that it is running
+ * 
+ * @return true if the connection was successfully established, otherwise false
+ */
+bool arduino_connect(char *device_file)
+{
+    // Open the serial device
+    device.set_device_file(device_file);
+    if (!device.ser_connect(BAUD_RATE)) return false;
+    device.ser_flush();
+    
+    printf("Waiting for Arduino...");
+    while (!(comm.check_for_message() && comm.received_message().id == MSG_ID_PING)) {
+        // Wait for ping
+    }
+    // There might be more ping messages sitting in the buffer, so flush them all out
+    device.ser_flush();
+    printf("Connected!\n");
+
+    return true;
+}
+
+/**
+ * @brief Disconnects from the Arduino
+ */
+void arduino_disconnect()
+{
+    device.ser_disconnect();
+}
+
+/**
+ * @brief Attempts to command the Arduino to move to the provided destination at the
+ *        provided velocity
+ * 
+ * @param dest_mm   Pointer to two floats (the absolute x and y coordinates in mm)
+ * @param vel_mm_s  Pointer to two floats (the x and y velocities in mm/s)
+ */
+void arduino_attempt_move(float *dest_mm, float *vel_mm_s)
```

Comment:
Would remove attempt, because in my mind every function attempts to do things haha, would just change it to `arduino_move`

Reply:
There's so many things that could go wrong with a move that it felt extra "attempty", so I added it to the name, but yeah I agree, every function is an attempt, so will remove